### PR TITLE
Fix locale-dependent data files

### PR DESCRIPTION
**Pull request by nick-l-o3de**
## What does this PR do?

Fix for locale-dependent data files being read/written
Fixes a variety of issues related to having a locale set on the machine that uses the comma as the decimal separator.
    
Two main approaches
* For AssetBuilder, always set, globally in the entire application and thus all builders, the "C" locale, which is the culture invariant locale
* For all other applications, create a PALified scoped thread-local locale setter that can be scoped around the parts where files are read or numbers are converted, then add it around all those sensitive places.
   * ObjectStream read and write
   * Json didn't need handling, as rapidjson already uses a culture invariant number formatter
   * Script Canvas read and write (including LUA generation)
   * LUA execution (lua will actually fail to work right if its not in the invariant locale)
   * Console variable execution (since it comes from a cfg file and that can be considered portable and invariant)
   * Added a new test for objectstream that verifies that the invariant locale is working backwards and forwards.
   *  Added a JSON test that verifies the same.

result:  Fixes issue #11139 and also issue #10987 and also issue #11680 and any similar that are to do with loading and saving.

Testing

Reproduced the issues in the above one before fixing it, they work fine after this fix.

Problem: I have no IOS, android, or mac devices to test on.  Theoretically, unixlike should be fine, as the documentaiton on Apple's developer website indicates that all these functions are available and work the same in the posix standard.   But how to test?

### CODE REVIEW

**Comment by nick-l-o3de** [comment]
May also fix https://github.com/o3de/o3de/pull/18171

File: Code/Framework/AzCore/AzCore/Math/MathScriptHelpers.cpp [hunk 1]
```diff
@@ -11,23 +11,30 @@
 #include <AzCore/Math/Vector4.h>
 #include <AzCore/Math/Quaternion.h>
 #include <AzCore/Math/Transform.h>
+#include <AzCore/Serialization/Locale.h>
 
 namespace AZ
 {
     AZStd::string Vector3ToString(const Vector3& v)
     {
+        AZ::Locale::ScopedSerializationLocale locale; // colors<--->string should be intepreted in the "C" Locale.
```
Comment (spham-amzn) [inline Code/Framework/AzCore/AzCore/Math/MathScriptHelpers.cpp:20]: did you mean `vectors<--->string`?

File: Code/Legacy/CrySystem/XML/xml.cpp [hunk 2]
```diff
@@ -26,30 +27,7 @@
...
-
-private:
-    char* m_oldLocale;
-};
-#define SCOPED_LOCALE_RESETTER LocaleResetter l
-#else
-// noop on Windows
-#define SCOPED_LOCALE_RESETTER
-#endif
+#define SCOPED_LOCALE_RESETTER AZ::Locale::ScopedSerializationLocale localeResetter;
```
Comment (spham-amzn) [inline Code/Legacy/CrySystem/XML/xml.cpp:30]: Do we need this #define anymore? Can we just use a local `localeResetter` instead?

**Review by galibzon — APPROVED** [review]

**Review by spham-amzn — APPROVED** [review]

File: Code/Legacy/CrySystem/XML/xml.cpp (hunk 2 above)
Comment (nick-l-o3de) [reply Code/Legacy/CrySystem/XML/xml.cpp:30]: I'll replace them all, it won't have an impact on the actual running.

File: Code/Framework/AzCore/AzCore/Math/MathScriptHelpers.cpp (hunk 1 above)
Comment (nick-l-o3de) [reply Code/Framework/AzCore/AzCore/Math/MathScriptHelpers.cpp:20]: I'll fix the comment.  Copy and paste!

**Comment by nick-l-o3de** [comment]
Turns out some of the symbols I used were not available on all unixlikes.  I did some timing and the cheapness of this call combined with the rare use of it (its not once per frame even, its once in a while when reading data files) means that the over-complexity was pointless.

**Comment by nick-l-o3de** [comment]
I had to rebase it on latest 2409, and then force push it to fix the multiplayer timeout bug :(
I'll give jenkins another try.

**Pull request merged**